compiler/packages/babel-plugin-react-compiler/src/Validation/ValidatePreservedManualMemoization.ts

```diff
@@ -258,6 +260,13 @@ function validateInferredDep(
     }
   }
   let errorDiagnostic: CompareDependencyResult | null = null;
```

Comment:
nit: move the `let errorDiagnostic ...` down since it's unrelated to the new if block